### ISSUE TITLE
DateEdit Not Binding on Server, Binds Okay in Debugger

### ISSUE THREAD

**Issue by cpurick** [Megabit/Blazorise#2635]
I have a grid, and an edit form below it that shows the selected item. Testers reported that changing the value on the form is not updating a newly added item in the grid. Upon close inspection, it appears we are not changing the bound object at all, just the date displayed in the control itself. Furthermore, this is only evident on the deployed application -- it runs normally in the browser against local IIS Express. In fact, it seems as if I almost cannot break it in IIS Express even if I try, and yet nothing I do will get it to run on the server.

This is the control:

```cshtml
<Validation Validator="ValidateRateStartDate">
    <Field Horizontal="true">
        <FieldLabel Class="field-label" Style="width:9rem;"> @localizer.Get( "Start Date" ):</FieldLabel>
        <FieldBody>
            <DateEdit Style="width:12rem;" TValue="DateTime" @bind-Date="@PageModel.SelectedEffectiveRateVM.RateEffectiveStart" ReadOnly="!(PageModel.SelectedEffectiveRateVM.IsEditable && PageModel.SelectedEffectiveRateVM.RateEffectiveStartDateEditable)" Min="@PageModel.AssignmentStartDate" />
        </FieldBody>
    </Field>
</Validation>
```

I have tried swapping out @bind-Date for a Date/DateChanged combination. I have also tried removing the Validator. The control was originally toggling from disabled to enabled, so I fixed that to use ReadOnly instead.

Every other control works. The DateEdit even works in the debugger. The only problem is that this one control is not changing the bound property when run on the server. Can you offer any suggestions?

**Comment by David-Moreira** [comment]
Hello,
Can you provide the blazorise version you are using? 
Also are you using Blazor Webassembly or blazor Server? 
We'll try to reproduce on our end, but generally, if you have the time a repo reproducing the error helps us a lot, thank you.

**Comment by cpurick** [comment]
We are on Bootstrap 0.9.3.6, Blazor Server.

**Comment by stsrki** [comment]
@cpurick What Blazorise provider you use?

**Comment by stsrki** [comment]
The only thing I can think of why it would break on the server is that you have different culture settings on your dev machine vs the server. Although it would be really strange why that would break it.

We would need more information and if possible a way to reproduce it before we could fix it.

**Comment by David-Moreira** [comment]
That's a pretty good point @stsrki makes and would explain different behaviour across environments, 
What is your server culture? Do you have a staging or testing environment where you can test setting the culture to something else? 
Or set the culture locally (IIS Express) to the same as the server.

**Comment by cpurick** [comment]
I can check the cultures. We interpret this as the events not firing when the user edits the date.

Do we believe the culture impact would be limited to date behavior, as I am seeing?

**Comment by David-Moreira** [comment]
Let's first track the problem... And only then evaluate the impact if that's indeed the problem. Dates and currencies/decimal parsing are the main ones when dealing with culture. 
Let us know your findings please.

**Comment by cpurick** [comment]
I cannot find explicit culture settings. Where would I check this? We're an international company and there is some localization in the code, but I believe everything I am doing here is US-oriented. What would I need to check to confirm?

**Comment by David-Moreira** [comment]
The microsoft docs should help:
https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.currentculture?view=net-5.0
Let me know if you have any doubts.

**Comment by cpurick** [comment]
That is useful and I understand it. How do I explicitly get server-side and client-side values? Will Blazor code always return the server culture?

**Comment by David-Moreira** [comment]
I'm not sure I follow your question. The culture is set on the server thread that's serving the request and can be manipulated if you wish. 
There's an example in the docs 

using System;
using System.Globalization;

public class Example
{
   public static void Main()
   {
      CultureInfo culture = CultureInfo.CurrentCulture;
      Console.WriteLine("The current culture is {0} [{1}]",
                        culture.NativeName, culture.Name);
   }
}
// The example displays output like the following:
//       The current culture is English (United States) [en-US]

**Comment by cpurick** [comment]
Sorry, I thought you cared about client-side culture. As I read it, all that matters is the difference between the server and the IIS Express experience, in which case both are US-en.

Also, we are wondering if a firewall is involved. Is there any particular network dependency associated with this control?

One thing we are noticing is that, when they work, the bindings on this control seem to function keystroke-by-keystroke, even when ChangeTextOnKeypress is false. Is this expected behavior?

**Comment by cpurick** [comment]
We did track this down to a firewall issue. Manipulating the date sends date strings in a request that includes a suspicious character which looks something like a 'U' with an accent mark. Our F5 was treating this like a malicious request. We started looking at the firewall after the standard InputDate control similarly failed to work. If I can get a capture of this I will submit it here for your knowledgebase.
